[1.6] Add PHP 8.3 support

GrahamCampbell opened the pull request.

GrahamCampbell committed:
Add PHP 8.3 support

divinity76 reviewed (approved):
https://github.com/PHPCompatibility/PHPCompatibility detects no relevant PHP8.3 compatibility issues.
LGTM 👍

GrahamCampbell commented:
> LGTM 👍

I don't know how you can say that, when the CI shows that the code is broken on PHP 8.3.

divinity76 commented:
> > LGTM 👍
> 
> I don't know how you can say that, when the CI shows that the code is broken on PHP 8.3.

somehow missed that, sorry.

Anyway, made a fix: https://github.com/chrome-php/wrench/pull/20

divinity76 committed:
PHP8.3 compatibility #2 (#20)

* PHP8.3 compatibility

Turns out PHP<8.3's proc_open does not really care if the requested working_dir does not exist, while PHP>=8.3's proc_open bails if it does not exist.

And due to a typo in the code, the dir never existed.

And due to insufficient error checking, the error was not really detected.

Took fking 4 hours to track down, and the fix is 1 byte. https://github.com/divinity76/wrench/pull/1

Should probably complain to the php-src bugtracker, i am guessing the proc_open behavior change was unintentional

* StyleCI

StyleCIBot committed:
Apply fixes from StyleCI

divinity76 committed:
PHP8.3 compatibility (#18)

strftime was deprecated in PHP8.1.0.
This is the only PHP8.3.0 compatibility issue detected by the PHPCompatibility scanner git version right now:

hans@DESKTOP-EE15SLU:~/projects/PHPCompatibility$ ./vendor/bin/phpcs ../wrench/ --standard=PHPCompatibility

FILE: /home/hans/projects/wrench/examples/StatusApplication.php
--------------------------------------------------------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
--------------------------------------------------------------------------------------------------------------------
 112 | WARNING | Function strftime() is deprecated since PHP 8.1; Use date() or IntlDateFormatter::format() instead
--------------------------------------------------------------------------------------------------------------------

Time: 710ms; Memory: 16MB

Related: https://github.com/chrome-php/wrench/pull/14 and https://github.com/chrome-php/chrome/issues/605
